tetris/src/classes/world3d.ts

```diff
@@ -0,0 +1,301 @@
+// }
+
+const GROUND_MATERIAL = "#ff7700";
+const GAMEFIELD_WIDTH = 4.0;
+const GAMEFIELD_COLOR = [255, 255, 255, 0.2];
+const CUBES_BORDER_WIDTH = 6.0;
+const CUBES_BORDER_COLOR = [255, 255, 255, 0.2];
+const LIGHT = {
+    POSITION: [2, 20, 2],
+    INTENSITY: 1
+}
+
+interface IMaterials {
+    [key: string]: BABYLON.StandardMaterial
+}
+
+export class World3d {
+    private readonly coloredMaterials: IMaterials;
+    private canvas: HTMLCanvasElement;
+    readonly scene: BABYLON.Scene;
+    private camera: BABYLON.ArcRotateCamera;
+    private light: BABYLON.HemisphericLight;
+    private ground: BABYLON.Mesh;
+    private gameField: BABYLON.Mesh;
+    private figure: BABYLON.Mesh[];
+    private finalFigure: BABYLON.Mesh[];
+    private heap: BABYLON.Mesh[];
+    private start: number;
+    public publisher3D = new Publisher({
+        getFigure: () => this.figure,
+        setFigure: (figure: BABYLON.Mesh[]) => this.figure = figure,
+        rerenderFigure: () => this.rerenderFigure(),
+        updateFigure: () => this.updateFigure(),
+        rerenderFinalFigure: () => this.rerenderFinalFigure(),
+        updateFinalFigure: () => this.updateFinalFigure(),
+        rerenderHeap: () => this.rerenderHeap(),
+        updateHeap: () => this.updateHeap(),
+        rerenderGround: () => this.rerenderGround(),
+        rerenderGameField: () => this.rerenderGameField(),
+        // moveCamera: () => this.moveCamera(),
+        updateGameState: () => this.updateGameState()
+    });
+
+    public constructor(canvas: HTMLCanvasElement, private publisherStore: Publisher<IListeners>) {
+        this.canvas = canvas;
+        const engine = new BABYLON.Engine(canvas);
+        this.scene = new BABYLON.Scene(engine);
+        this.camera = this.createCamera(canvas);
+        this.light = this.createLight();
+        this.coloredMaterials = this.createMaterials();
+        this.ground = this.createGround();
+        this.gameField = this.createGameField();
+        this.publisherStore.dispatch("createNewFigure")();
+        this.figure = this.createFigureFromState();
+        this.finalFigure = this.createFinalFigureFromState();
+        this.heap = this.createHeapFromState();
+        this.start = Date.now().valueOf();
+
+        canvas.addEventListener("resize", () => {
+            engine.resize();
+        });
+
+        this.scene.registerBeforeRender(() => {
+            const progress = Date.now().valueOf() - this.start;
+            if (progress >= this.publisherStore.get("getState").delay.current && this.publisherStore.get("getState").gameState === IGameState.Playing) {
+                this.start = Date.now().valueOf();
+                this.publisherStore.dispatch("moveFigure")(Axis.Y, -this.publisherStore.get("getState").dy);
+                this.publisherStore.dispatch("deleteLevels")();
+            }
+        });
+
+        engine.runRenderLoop(() => {
+            this.scene.render();
+        });
+    }
+
+    rerenderFigure() {
```

Comment:
please add property accessors for each method of class (`public`, `private` or `protected`)